src/rubrix/client/datasets.py

```diff
@@ -802,44 +809,65 @@ def prepare_for_training(self) -> "datasets.Dataset":
 
 
         """
-        import datasets
+        if framework is None:
 
-        has_annotations = False
-        for rec in self._records:
-            if rec.annotation is not None:
-                has_annotations = True
-                break
-
-        if not has_annotations:
-            return datasets.Dataset.from_dict({})
-
-        class_tags = ["O"]
-        class_tags.extend(
-            [
-                f"{pre}-{label}"
-                for label in sorted(self.__all_labels__())
-                for pre in ["B", "I"]
-            ]
-        )
-        class_tags = datasets.ClassLabel(names=class_tags)
+            import datasets
 
-        def spans2iob(example):
-            r = TokenClassificationRecord(
-                text=example["text"],
-                tokens=example["tokens"],
-                annotation=self.__entities_to_tuple__(example["annotation"]),
+            has_annotations = False
+            for rec in self._records:
+                if rec.annotation is not None:
+                    has_annotations = True
+                    break
+
+            if not has_annotations:
+                return datasets.Dataset.from_dict({})
+
+            class_tags = ["O"]
+            class_tags.extend(
+                [
+                    f"{pre}-{label}"
+                    for label in sorted(self.__all_labels__())
+                    for pre in ["B", "I"]
+                ]
             )
-            return class_tags.str2int(r.spans2iob(r.annotation))
+            class_tags = datasets.ClassLabel(names=class_tags)
 
-        ds = (
-            self.to_datasets()
-            .filter(self.__only_annotations__)
-            .map(lambda example: {"ner_tags": spans2iob(example)})
-        )
-        new_features = ds.features.copy()
-        new_features["ner_tags"] = [class_tags]
+            def spans2iob(example):
+                r = TokenClassificationRecord(
+                    text=example["text"],
+                    tokens=example["tokens"],
+                    annotation=self.__entities_to_tuple__(example["annotation"]),
+                )
+                return class_tags.str2int(r.spans2iob(r.annotation))
+
+            ds = (
+                self.to_datasets()
+                .filter(self.__only_annotations__)
+                .map(lambda example: {"ner_tags": spans2iob(example)})
+            )
+            new_features = ds.features.copy()
+            new_features["ner_tags"] = [class_tags]
+
+            return ds.cast(new_features)
+
+        elif framework == "spacy":
+
+            nlp = spacy.blank("en")
+
+            db = DocBin()
```

Comment:
Can the `attrs`  init argument be left with the default value? See the [spaCy docs](https://spacy.io/api/docbin#init)

Reply:
Do we have any information for filling up the `attrs`? I've checked the attributes list but I don't know if we can extract some information from our dataset to not leave that empty.

Reply:
Nop. I have no idea about that. Just asking you if you have a suggestion. We can keep the defaults for now

Reply:
I also have no idea about it, but from my point of view, there is nothing that we can directly relate to. Maybe open an issue to look into further?

Reply:
> @frascuchon Should I change `if framework is None:` for `if framework == "hf": ` and make HF default?

Not sure if `hf` is quite descriptive,`transformers`? `huggingface`?  Maybe @dcfidalgo or @dvsrepo can answer this.